src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,810 @@
+                        stmt.args,
+                    ),
+                    .create_transfers => try create(
+                        tb.Transfer,
+                        .transfer,
+                        context,
+                        arena,
+                        stmt.args,
+                    ),
+                    .lookup_transfers => try lookup(
+                        "transfer",
+                        context,
+                        arena,
+                        stmt.args,
+                    ),
+                }
+                return;
+            }
+
+            // No input was parsed.
+            context.debug("No command was parsed, continuing.\n", .{});
+        }
+
+        fn repl(
+            context: *Context,
+            arena: *std.heap.ArenaAllocator,
+        ) !void {
+            print("> ", .{});
+
+            const in = std.io.getStdIn();
+            var stream = std.io.bufferedReader(in.reader()).reader();
+
+            var input: []u8 = undefined;
+
+            if (stream.readUntilDelimiterOrEofAlloc(
+                arena.allocator(),
+                ';',
+                max_single_repl_input,
+            )) |maybe_bytes| {
+                if (maybe_bytes) |bytes| {
+                    input = bytes;
+                } else {
+                    // EOF.
+                    context.event_loop_done = true;
+                    context.fail("\nExiting.\n", .{});
+                    return;
+                }
+            } else |e| {
+                context.event_loop_done = true;
+                fail("Failed to read from stdin: {any}\n", .{e});
+                return e;
+            }
+
+            var stmt = Parse.parse_statement(context, arena, input) catch return;
+            try do_statement(
+                context,
+                arena,
+                stmt,
+            );
+        }
+
+        fn display_help() void {
+            const version = build_options.git_tag orelse "experimental";
+            print("TigerBeetle CLI Client " ++ version ++ "\n" ++
+                \\  Hit enter after a semicolon to run a command.
+                \\
+                \\Examples:
+                \\  create_accounts id=1 code=10 ledger=700,
+                \\                  id=2 code=10 ledger=700;
+                \\  create_transfers id=1 debit_account_id=1
+                // This should show up on a single line to demonstrate
+                // a long line to the user. However, long lines make our
+                // linter unhappy.
+            ++ " credit_account_id=2 amount=10 ledger=700 code=10;" ++
+                \\  lookup_accounts id=1;
+                \\  lookup_accounts id=1, id=2;
+                \\
+                \\
+            , .{});
+        }
+
+        pub fn run(
+            arena: *std.heap.ArenaAllocator,
+            addresses: []std.net.Address,
+            statements: ?[]const u8,
+            verbose: bool,
```

Comment:
Ah that makes sense. However, no one is calling it with anything but arguments from the client directly.

So it would become:

```
.run(a, .{.statements = args.statements, .verbose = args.verbose})
```

Which only makes this code more verbose without really adding anything?

Reply:
Ah, good point! Never mind then.